### ISSUE TITLE
Make the title h-tag of ProductiveCard and ExpressiveCards more flexible for accessibility.

### ISSUE THREAD

**Issue by alexandria-lewis** [carbon-design-system/ibm-products#2258]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## What did you want to know?

Assuming there is other content inside one of these cards, with their own titles, it would require that the title have a different h-tag. Because the numbers matter for accessibility. Can there be an option to choose which h-tag we want to use on our cards. It doesn't have to affect the style, it's for accessibility and not locking this component at the bottom of content hierarchy. 

Use of the proper h-tag in the proper order is a bare minimum requirement for WCAG.

## Relevant information
[https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-cards-expressivecard--default](https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-cards-expressivecard--default)

[https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-cards-productivecard--default](https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-cards-productivecard--default)

[https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=242%2C246%2C132#meaningful-sequence](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=242%2C246%2C132#meaningful-sequence)

**Comment by glapadre** [comment]
@alexandria-lewis the default h-tag is a <h6>, are you looking for an option in the controls that would allow you to switch between a h1, h2, h3, ect.. for the title? For accessibility reasons I understand the ask but my concern is it might get confusing since we already have an option for titleSize (which changes the style sizing 'default' or 'large'. So there would be two controls, one specifically to update the element tag (with no styling) - this would be new **ex.  h1, h2, h3, ect..,** and the current control  `titleSize` which would actually change the size of the text.

**Comment by glapadre** [comment]
@alexandria-lewis any update on the above?

**Comment by alexandria-lewis** [comment]
@glapadre Hi sorry for the late response. I understand it would be confusing with the size changes already being handled. But if you labeled it in such a way as to indicated it's a tag control for accessibility maybe that would separate it more? It's unfortunate that title tags are often used for their styles, but they are more importantly for traversing the accessibility tree.

So maybe naming the header tag option in such a way that decouples it from style is possible?

**Comment by elycheea** [comment]
The PR #7284 adds Carbon’s `Section` and [`Heading`](https://react.carbondesignsystem.com/?path=/docs/components-heading--overview#overview) which now infer the appropriate heading level.
